distributed_nose/plugin.py

```diff
@@ -137,33 +150,51 @@ def configure(self, options, config):
                 # are 1-indexed and this prevents the need to do
                 # offsetting when we access this structure with
                 # a node-number elsewhere.
-                self.lpt_nodes = [
-                    {
-                        'processing_time': 0,
-                        'classes': set()
-                    }
-                    for _ in range(self.node_count + 1)
-                ]
 
                 with open(self.lpt_data_filepath) as f:
-                    self.lpt_data = json.load(f)
-
-                    # for now, lpt only operates at the class level
-                    self.hash_by_class = True
 
-                    sorted_lpt_data = sorted(
-                        self.lpt_data.items(),
-                        key=lambda t: t[1]['duration'],
-                        reverse=True
-                    )
-
-                    for c, data in sorted_lpt_data:
-                        node = min(
-                            self.lpt_nodes[1:],
-                            key=lambda n: n['processing_time']
+                    self.lpt_data = json.load(f)
+                    self.lpt_nodes = [
+                        {
+                            'processing_time': 0,
+                            'objects': set()
+                        }
+                        for _ in range(self.node_count + 1)
+                    ]
+
+                    if self.distribution_type == "test":
+
+                        # The data for test looks like this:
```

Comment:
wondering if it might be worthwhile to make the data formats more similar, so these two branches don't have code that's only slightly different?

Reply:
not sure, because we need O(1) access while checking if we need to run this particular test at this node:
```return full_name in self.lpt_nodes[self.node_id]['objects']```
so for class distribution we need a hash table with classnames, for test distribution we need hasg table with test names

Reply:
I can probably create it during execution using data for class distribution...hm..

Reply:
but it won't affect the code much, so similiar code will remain I believe

Reply:
Yeah, I think I figured it out